src/proof.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::nibbleslice::NibbleSlice;
+use crate::node::Node;
+use ccrypto::{blake256, BLAKE_NULL_RLP};
+use primitives::Bytes;
+use primitives::H256;
+
+// Unit of a proof.
+//#[derive(Clone, Eq, PartialEq, Debug, RlpEncodable, RlpDecodable)]
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct CryptoProofUnit<H, K, V> {
```

Comment:
This crate is `rust-merkle-trie`. rust-merkle-trie crate does not need to provide a generic Proof method.
What do you think about using concrete types here, and creating generic types in ICS code?

@junha1 @sgkim126

Reply:
https://github.com/CodeChain-io/rust-merkle-trie/pull/15#discussion_r374474637